### DISCUSSION TITLE
Undefined constant "activeTab" x-filament::badge :icon

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#12594] [Help]
### Package

Other

### Package Version

v3.2.71

### How can we help you?

once I remove `:` from icon I get this error:

`Svg by name "activeTab === 'monthly' ? 'heroicon-o-calendar-days' : 'heroicon-o-calendar'" from set "default" not found.
`
once I return it to `:icon` I get Undefined constant **activeTab**

```
<x-filament-panels::page>
        <div x-data="{ activeTab: 'monthly' }">
            ...

            <div>
                <div>
                    <div>
                        <div>
                            <h1>
                            </h1>
                            <x-filament::badge
                                :icon="activeTab === 'monthly' ? 'heroicon-o-calendar-days' : 'heroicon-o-calendar'"
                            />
                        </div>
                    </div>
                </div>
                </div>
            </div>
        </div>
</x-filament-panels::page>
```

**Comment by xalabama** [comment]
This error occurs because you try to use javascript inside php context, so php dosn't know something about your definition and try to access a constant. But simply you can help the livewire component you want to load to understand this if you using a double : as prefix of your param.

```
<x-filament::badge
    ::icon="activeTab === 'monthly' ? 'heroicon-o-calendar-days' : 'heroicon-o-calendar'"
/>
```

**Reply by devhoussam1998** [reply]
using `::icon="activeTab === 'monthly' ? 'heroicon-o-calendar-days' : 'heroicon-o-calendar'"` icon not been displayed anymore

**Reply by xalabama** [reply]
May be a look in the docs will helps you https://filamentphp.com/docs/3.x/support/blade-components/badge
